### PR TITLE
[TS Migration] Migrate Icon to TypeScript

### PR DESCRIPTION
**Pull request by JKobrynski**
### Fixed Issues

$ #25157 
PROPOSAL: N/A

### Tests

1. Log into the app
2. Look at icons in the RHP and make sure they look fine

- [x] Verify that no errors appear in the JS console

### Offline tests

N/A

### QA Steps

Same as **Tests** steps above

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

android

</details>

<details>
<summary>iOS: Native</summary>

ios

</details>

<details>
<summary>iOS: mWeb Safari</summary>

safari

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

web

</details>

<details>
<summary>MacOS: Desktop</summary>

desktop

</details>

### CODE REVIEW

File: src/components/Icon/index.tsx [hunk 1]
```diff
@@ -1,73 +1,81 @@
...
-const defaultProps = {
-    width: variables.iconSizeNormal,
-    height: variables.iconSizeNormal,
-    fill: themeColors.icon,
-    small: false,
-    inline: false,
-    additionalStyles: [],
-    hovered: false,
-    pressed: false,
+    additionalStyles?: ViewStyle[];
```
Comment (pasyukevich) [inline src/components/Icon/index.tsx:43]: Can we use here` StyleProp<ViewStyle>` ?

File: src/components/Icon/index.tsx [hunk 2]
```diff
@@ -1,73 +1,81 @@
...
 
     /** Is icon hovered */
-    hovered: PropTypes.bool,
+    hovered?: boolean;
 
     /** Is icon pressed */
-    pressed: PropTypes.bool,
+    pressed?: boolean;
 
     // eslint-disable-next-line react/forbid-prop-types
```
Comment (pasyukevich) [inline src/components/Icon/index.tsx:42]: Looks like it is redundant

File: src/components/Icon/index.tsx [hunk 3]
```diff
@@ -1,73 +1,81 @@
...
+        additionalStyles: [],
+        hovered: false,
+        pressed: false,
+    };
+
     render() {
         const width = this.props.small ? variables.iconSizeSmall : this.props.width;
         const height = this.props.small ? variables.iconSizeSmall : this.props.height;
-        const iconStyles = [StyleUtils.getWidthAndHeightStyle(width, height), IconWrapperStyles, styles.pAbsolute, ...this.props.additionalStyles];
+        const iconStyles = [StyleUtils.getWidthAndHeightStyle(width ?? 0, height), IconWrapperStyles, styles.pAbsolute, ...(this.props.additionalStyles ?? [])];
```
Comment (kubabutkiewicz) [inline src/components/Icon/index.tsx:64]:
```suggestion
        const iconStyles = [StyleUtils.getWidthAndHeightStyle(width, height), IconWrapperStyles, styles.pAbsolute, ...(this.props.additionalStyles ?? [])];
```
We have a default value for `width` so I guess we can skip `??` in this case

File: src/components/Icon/index.tsx [hunk 4]
```diff
@@ -1,73 +1,81 @@
...
 
         if (this.props.inline) {
             return (
                 <View
                     testID={`${this.props.src.name} Icon`}
-                    style={[StyleUtils.getWidthAndHeightStyle(width, height), styles.bgTransparent, styles.overflowVisible]}
+                    style={[StyleUtils.getWidthAndHeightStyle(width ?? 0, height), styles.bgTransparent, styles.overflowVisible]}
```
Comment (kubabutkiewicz) [inline src/components/Icon/index.tsx:70]: Same comment

File: src/components/Icon/svgs/LoungeAccessIcon.tsx [hunk 5]
```diff
@@ -1,26 +1,25 @@
...
 
-const defaultProps = {
+const defaultProps: LoungeAccessIconProps = {
     fill: themeColors.icon,
     hovered: 'false',
     pressed: 'false',
 };
 
-function LoungeAccessIcon(props) {
+function LoungeAccessIcon(props: LoungeAccessIconProps) {
```
Comment (kubabutkiewicz) [inline src/components/Icon/svgs/LoungeAccessIcon.tsx:22]: maybe we can destructure props here

File: src/components/Icon/svgs/LoungeAccessIcon.tsx [hunk 6]
```diff
@@ -1,26 +1,25 @@
...
-    fill: PropTypes.string,
+    fill?: string;
 
     /** Is icon hovered */
-    hovered: PropTypes.string,
+    hovered?: string;
 
     /** Is icon pressed */
-    pressed: PropTypes.string,
+    pressed?: string;
```
Comment (kubabutkiewicz) [inline src/components/Icon/svgs/LoungeAccessIcon.tsx:13]: I know that it was like that before , but this seems super od for me , why its string? 😄

File: src/components/Icon/index.tsx (hunk 1 above)
Comment (JKobrynski) [reply src/components/Icon/index.tsx:43]: It's always an array, so I guess not this time, hence the name `additionalStyle**s**`

File: src/components/Icon/index.tsx (hunk 2 above)
Comment (JKobrynski) [reply src/components/Icon/index.tsx:42]: Right

File: src/components/Icon/index.tsx (hunk 4 above)
Comment (JKobrynski) [reply src/components/Icon/index.tsx:70]: Same here 😄

File: src/components/Icon/svgs/LoungeAccessIcon.tsx (hunk 5 above)
Comment (JKobrynski) [reply src/components/Icon/svgs/LoungeAccessIcon.tsx:22]: Great idea!

File: src/components/Icon/index.tsx (hunk 1 above)
Comment (pasyukevich) [reply src/components/Icon/index.tsx:43]: According to our guide, we should always use it without an array wrapper - https://github.com/Expensify/App/blob/main/contributingGuides/TS_CHEATSHEET.md#style-props

`StyleProp` already considers this

Comment (JKobrynski) [reply src/components/Icon/index.tsx]: Updated!

File: src/components/Icon/IconWrapperStyles/index.ios.ts
Comment (blazejkustra) [inline src/components/Icon/IconWrapperStyles/index.ios.ts:1]: Aad types.ts file with a common type for these files.

File: src/components/Icon/svgs/LoungeAccessIcon.tsx [hunk 7]
```diff
@@ -30,31 +29,37 @@ function LoungeAccessIcon(props) {
...
                 />
                 <Polygon
+                    // @ts-expect-error - react-native-web property
                     className="st1"
                     points="28.5,7 28.5,7 28.5,7  "
                 />
             </G>
             <G>
                 <Path
+                    // @ts-expect-error - react-native-web property
```
Comment (blazejkustra) [inline src/components/Icon/svgs/LoungeAccessIcon.tsx:62]: Can you try to augment the `react-native-svg` type? `src/types/modules/`

**Review by blazejkustra — CHANGES_REQUESTED** [review]

**Comment by blazejkustra** [comment]
Try to add some QA steps! FYI tests are failing

File: src/components/Icon/IconWrapperStyles/index.ios.ts
Comment (JKobrynski) [reply src/components/Icon/IconWrapperStyles/index.ios.ts:1]: Done!

File: src/components/Icon/svgs/LoungeAccessIcon.tsx (hunk 7 above)
Comment (JKobrynski) [reply src/components/Icon/svgs/LoungeAccessIcon.tsx]: Great idea! I pushed a solution, let me know what you think 😄

File: src/components/Icon/svgs/LoungeAccessIcon.tsx [hunk 8]
```diff
@@ -33,8 +32,9 @@ function LoungeAccessIcon(props) {
                 enableBackground: 'new 0 0 40 40',
             }}
             xmlSpace="preserve"
-            // eslint-disable-next-line react/jsx-props-no-spreading
-            {...props}
```
Comment (blazejkustra) [inline src/components/Icon/svgs/LoungeAccessIcon.tsx:37]: Make sure no other props were passed before

File: src/types/modules/react-native-svg.d.ts [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+    }
+
+    // eslint-disable-next-line @typescript-eslint/consistent-type-definitions
+    interface SvgProps extends BaseSvgProps {
+        xmlns?: string;
+        xmlnsXlink?: string;
+        xmlSpace?: string;
+        style?: StyleProp<SvgStyle>;
+        hovered?: string;
+        pressed?: string;
```
Comment (blazejkustra) [inline src/types/modules/react-native-svg.d.ts:17]: Why all these props are needed?

File: src/types/modules/react-native-svg.d.ts [hunk 10]
```diff
@@ -0,0 +1,24 @@
+import {StyleProp} from 'react-native';
+import {CommonPathProps as BaseCommonPathProps, SvgProps as BaseSvgProps} from 'react-native-svg';
+
+declare module 'react-native-svg' {
+    // eslint-disable-next-line @typescript-eslint/consistent-type-definitions
+    interface SvgStyle extends ViewStyle {
```
Comment (blazejkustra) [inline src/types/modules/react-native-svg.d.ts:6]: Should `SvgStyle` extend `ViewStyle`?

**Review by blazejkustra — CHANGES_REQUESTED** [review]

File: src/components/Icon/svgs/LoungeAccessIcon.tsx (hunk 8 above)
Comment (JKobrynski) [reply src/components/Icon/svgs/LoungeAccessIcon.tsx:37]: Great point, turned out I missed some 😬

File: src/types/modules/react-native-svg.d.ts (hunk 9 above)
Comment (JKobrynski) [reply src/types/modules/react-native-svg.d.ts]: For some reason, when I augmented `react-native-svg` it started complaining about these values and them being undefined. If you remove the style prop, it starts showing errors related to these values, even without augmentation. I know it's weird. Do you have any ideas for possible causes?

File: src/types/modules/react-native-svg.d.ts (hunk 10 above)
Comment (JKobrynski) [reply src/types/modules/react-native-svg.d.ts]: It wasn't totally accurate, I modified it a bit

**Comment by JKobrynski** [comment]
@blazejkustra applied some changes, also fixed unit tests 😄

File: src/components/Icon/svgs/LoungeAccessIcon.tsx [hunk 11]
```diff
@@ -64,7 +72,6 @@ function LoungeAccessIcon(props) {
 }
 
 LoungeAccessIcon.displayName = 'LoungeAccessIcon';
-LoungeAccessIcon.propTypes = propTypes;
 LoungeAccessIcon.defaultProps = defaultProps;
```
[the remark this answers is not in the document]
Comment (JKobrynski) [reply src/components/Icon/svgs/LoungeAccessIcon.tsx]: Not at all, good point!

File: src/components/Icon/svgs/LoungeAccessIcon.tsx [hunk 12]
```diff
@@ -1,26 +1,31 @@
...
+
+    /** Icon's width */
+    width?: number;
+
+    /** Icon's height */
+    height?: number;
 };
 
-const defaultProps = {
+const defaultProps: LoungeAccessIconProps = {
```
[the remark this answers is not in the document]
Comment (JKobrynski) [reply src/components/Icon/svgs/LoungeAccessIcon.tsx]: Right!

File: src/components/Icon/__mocks__/Expensicons.ts [hunk 13]
```diff
@@ -0,0 +1,7 @@
+const Expensicons = jest.requireActual('../Expensicons');
+
+module.exports = Object.keys(Expensicons).reduce((prev, curr) => {
+    const fn = () => '';
```
[the remark this answers is not in the document]
Comment (JKobrynski) [reply src/components/Icon/__mocks__/Expensicons.ts:6]: My bad, I brought it back

**Review by BartoszGrajdek — APPROVED** [review]

**Comment by roryabraham** [comment]
Assigned to @akinwale for C+ review

**Comment by akinwale** [comment]
@JKobrynski Minor nitpick, the icons are in the RHP, not the LHN. Could you please update your test steps?

**Comment by akinwale** [comment]
TypeScript Checks task is failing. Please fix.

**Comment by JKobrynski** [comment]
@akinwale description updated, typechecks fixed!

**Comment by akinwale** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>

30579-ios-native

</details>

<details>
<summary>iOS: mWeb Safari</summary>

30579-ios-safari

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

30579-web

</details>

<details>
<summary>MacOS: Desktop</summary>

30579-desktop

</details>

**Review by akinwale — APPROVED** [review]
LGTM.

File: src/components/Icon/index.tsx [hunk 14]
```diff
@@ -83,15 +91,12 @@ class Icon extends PureComponent {
...
                 />
             </View>
         );
     }
 }
 
-Icon.propTypes = propTypes;
-Icon.defaultProps = defaultProps;
-
 export default Icon;
```
Comment (roryabraham) [inline src/components/Icon/index.tsx:102]: Missing `displayName` property

**Review by roryabraham — APPROVED** [review]

**Pull request merged**